Doc/whatsnew/3.11.rst

```diff
@@ -493,6 +494,222 @@ Optimizations
+| Subscript     | ``o[i]``           | Subscripting container types such as ``list``,        | 10-30%            | Irit Katriel,     |
+|               |                    | ``tuple`` and ``dict`` directly index the underlying  |                   | Mark Shannon      |
+|               |                    | data structures. Subscripting custom ``__getitem__``  |                   |                   |
+|               |                    | is also inlined similar to :ref:`inline-calls`.       |                   |                   |
++---------------+--------------------+-------------------------------------------------------+-------------------+-------------------+
+| Store         | ``o[i] = z``       | Similar to subscripting specialization above.         | ?                 |                   |
+| subscript     |                    |                                                       |                   |                   |
++---------------+--------------------+-------------------------------------------------------+-------------------+-------------------+
+| Calls         | ``f(arg)``         | Calls to common builtin (C) functions such as ``len`` | 20%               | Mark Shannon,     |
+|               |                    | and ``isinstance`` directly call their underlying C   |                   | Ken Jin           |
+|               | ``T(arg)``         | version. This avoids going through the internal       | 170%              |                   |
+|               |                    | calling convention.                                   |                   |                   |
+|               |                    |                                                       |                   |                   |
+|               |                    | Calls to certain Python functions are inlined similar |                   |                   |
+|               |                    | to :ref:`inline-calls`.                               |                   |                   |
+|               |                    |                                                       |                   |                   |
++---------------+--------------------+-------------------------------------------------------+-------------------+-------------------+
+| Load          | ``print; len``     | The object's index in the globals/builtins namespace  | 0% [1]_           | Mark Shannon      |
+| global        |                    | is cached. Loading globals and builtins require       |                   |                   |
+| variable      |                    | zero namespace lookups.                               |                   |                   |
++---------------+--------------------+-------------------------------------------------------+-------------------+-------------------+
+| Load          | ``o.attr``         | Similar to loading global variables. The attribute's  | 0% [2]_           | Mark Shannon      |
```

Comment:
We get a bigger speed up here, largely thanks to the virtual object dictionaries. More specializations also help here.
Don't have numbers though.